src/main/java/com/linecorp/armeria/common/ServiceInvocationContext.java

```diff
@@ -459,10 +461,18 @@ public void rejectPromise(Promise<?> promise, Throwable cause) {
             return;
         }
 
-        if (!(promise.cause() instanceof TimeoutException)) {
-            // Log reject failure unless it is due to a timeout.
-            logger().warn("Failed to reject a promise ({}) with {}", promise, cause, cause);
+        final Throwable firstCause = promise.cause();
+        if (firstCause instanceof TimeoutException) {
+            // Timed out already.
+            return;
+        }
+
+        if (Exceptions.isExpected(cause)) {
+            // The cause arrived later than firstCause was just a usual expected exception.
```

Comment:
// The exception that was thrown after firstCause (usually a transport-layer exception) was a usual expected exception, not an error.

I think